storage/src/main/java/tech/pegasys/teku/storage/store/Store.java

```diff
@@ -915,16 +832,20 @@ public boolean containsBlock(final Bytes32 blockRoot) {
     public SafeFuture<Optional<BeaconState>> retrieveCheckpointState(Checkpoint checkpoint) {
       BeaconState inMemoryCheckpointBlockState = block_states.get(checkpoint.getRoot());
       if (inMemoryCheckpointBlockState != null) {
-        return SafeFuture.completedFuture(
-            Optional.of(regenerateCheckpointState(checkpoint, inMemoryCheckpointBlockState)));
+        // Not executing the task via the task queue to avoid caching the result before the tx is
```

Comment:
I'm a bit confused with how you accomplish this:

Reply:
Hmm you just don't use the task queue and you just don't cache it as a result. Makes sense.